### PR TITLE
Add more implementation & proofs for trusty API

### PR DESCRIPTION
**Pull request by LinerSu**
- Add more unit proofs: `wait_any`, `accept`, `close`, `get_msg`, `put_msg`, `send_msg`, and `read_msg`.
- Update implementation of API (for `accept`, `connect`)
- Fix post conditions declared in `connect` proof.

### CODE REVIEW

File: seahorn/trusty/user/base/lib/libc-trusty/sea_handle_table.c [hunk 1]
```diff
@@ -212,6 +215,15 @@ handle_t sea_ht_new_port(bool secure) {
   return INVALID_IPC_HANDLE;
 }
 
+handle_t sea_ht_math_port(const char *path) {
+#define CASE(X)                                                                \
+  if ((strcmp(PHANDLE(X, path), path) == 0) && PHANDLE(X, active))             \
```
Comment (agurfinkel) [inline seahorn/trusty/user/base/lib/libc-trusty/sea_handle_table.c]: this is too expensive. we can't have this in the model. no `memcmp` and definitely no `strcmp`.

If you need paths, restrict them to be one character length

**Review by agurfinkel — CHANGES_REQUESTED** [review]
LGTM overall. 

The use of strcmp should be replaced by checking only the first character of the string.
Probably the input path string should be copied (again, only its first character)
`cex` fails in CI, needs to be addressed. Maybe using `smtfd` or `explicit_sp0` or both.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        verify_flag: ["", "--vac", "--cex"]
+        verify_flag: ["", "--vac", "--cex --horn-explicit-sp0=true"]
```
Comment (agurfinkel) [inline .github/workflows/main.yml]: we do not need `explicit-sp0` for all jobs. Only for some. Please add it only when necessary. It under approximates some behaviours. I am still unsure whether using it all the time will not miss some bugs.

**Review by agurfinkel — CHANGES_REQUESTED** [review]
LGTM

Only required change is to not enable explicit sp0 globally

**Comment by LinerSu** [comment]
Fixed it by splitting `ctest` under `cex` mode.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -27,5 +27,15 @@ jobs:
       - name: Docker 
         run: docker build -t verify-trusty . --file docker/seahorn-dev10-verify-trusty.Dockerfile
       
-      - name: Run Tests 
-        run: docker run -t verify-trusty /bin/bash -c "cd build && mkdir -p /tmp/verifyTrusty && env VERIFY_FLAGS=\"${{ matrix.verify_flag }}\" ctest --output-on-failure --timeout 2000 ."
+      - name: Get number of Cores
+        id: cores
+        run: echo "::set-output name=num_cores::$(nproc --all)"
+      
+      - if: ${{ matrix.verify_flag == '--cex' }}
```
Comment (agurfinkel) [inline .github/workflows/main.yml]: no. this only applies in CI. We specifically have local sea.yaml files to change options per individual proof. Please use these.

**Review by agurfinkel — CHANGES_REQUESTED** [review]

**Comment by LinerSu** [comment]
DONE!!!

**Review by agurfinkel — APPROVED** [review]

**Pull request merged**